backend/src/data/repositories/group/group.repository.ts

```diff
@@ -0,0 +1,54 @@
+import { Group as GroupM } from '~/data/models/models';
+import { Group as GroupEntity } from '~/services/group/group.entity';
+
+type Constructor = {
+  GroupModel: typeof GroupM;
+};
+
+class Group {
+  #GroupModel: typeof GroupM;
+
+  constructor({ GroupModel }: Constructor) {
+    this.#GroupModel = GroupModel;
+  }
+
+  async getGroupByNameAndTenant(
+    name: string,
+    tenantId: string,
+  ): Promise<GroupEntity | null> {
+    const group = await this.#GroupModel
+      .query()
+      .select()
+      .where({ name })
+      .andWhere({ tenantId })
+      .first();
+
+    if (!group) {
+      return null;
+    }
+
+    return Group.modelToEntity(group);
+  }
+
+  async create(group: GroupEntity): Promise<GroupM> {
```

Comment:
Repository should not return Model. It should return Entity

Reply:
But in the master repository create method returns also model:
![image](https://user-images.githubusercontent.com/86227025/153706377-3c6b1c05-11cb-4ef6-9dfa-f825f34256ce.png)

Reply:
> Repository should not return Model. It should return Entity

But in the master repository create method returns also model:
![image](https://user-images.githubusercontent.com/86227025/153706415-fc2ec8d3-5835-4565-a131-42a4e96ac70d.png)